schema/resolvers.js

```diff
@@ -263,12 +274,35 @@ const Resolvers = {
   },
 
   RoutingConditionValue: {
-    __resolveType: () => "IDArrayValue"
+    __resolveType: async ({ conditionId }, ctx) => {
+      const answerType = await ctx.repositories.Routing.getAnswerTypeByConditionId(
+        conditionId,
+        ctx
+      );
+      if (answerType === "Radio") {
+        return "IDArrayValue";
+      } else if (includes(["Currency", "Number"], answerType)) {
+        return "NumberValue";
+      }
+    }
   },
 
   IDArrayValue: {
-    value: ({ conditionId }, args, ctx) =>
-      ctx.repositories.Routing.findAllRoutingConditionValues({ conditionId })
+    value: async ({ conditionId }, args, ctx) => {
+      const conditionValues = await ctx.repositories.Routing.findAllRoutingConditionValues(
+        {
+          conditionId
+        }
+      );
+      return conditionValues.map(conditionValue => conditionValue.optionId);
+    }
+  },
+
+  NumberValue: {
+    id: async ({ conditionId }, args, ctx) =>
```

Comment:
This looks odd - we are making a query for each field instead of picking off an already exiting entity. Maybe I'm missing something.